[CXX] Do not initialize ENABLE_TESTS with the auto-downloaded C Driver (#135)
mongodb/docs-cpp: source/upgrade.txt

  .. note::

     This does not affect users who compile with C++17 or newer and have not set
     ``BSONCXX_POLY_USE_IMPLS=ON``

> .. _version-4.1-enable-tests-config:

Reviewer: S: I don't think you can add ref anchors within a section, so looks like this one just goes to :ref:`version-4.1-build-system-breaking`. you can remove it and replace reference to it with the general 4.1 breaking changes ref

> - Sets the ``ENABLE_TESTS`` configuration option to ``OFF`` by default for
>   auto-downloaded versions of the {+driver-short+}. :ref:`This is an expansion of the
>   change made in v4.0. <version-4.0-enable-tests-config>`

Reviewer: I: you can move this to the "Build System Breaking Changes in v4.1" section above
Also I think the problem is with the auto-generated C version that comes with the C++ driver, and this was setting ENABLE_TESTS to ON incorrectly: 
```suggestion
- Sets the ``ENABLE_TESTS`` configuration option to ``OFF`` by default for
  auto-downloaded versions of the C driver. :ref:`This is an expansion of the
  change made in v4.0. <version-4.0-enable-tests-config>`
```

  .. note::

     This does not affect users who compile with C++17 or newer and have not set
     ``BSONCXX_POLY_USE_IMPLS=ON``

.. _version-4.1-enable-tests-config:

> - Sets the ``ENABLE_TESTS`` configuration option to ``OFF`` by default,
>   including for auto-downloaded versions of the {+driver-short+}. To
>   re-enable building test targets, you must set ``ENABLE_TESTS`` to ``ON``.
>   To include test targets in the "all" target, you must also set the
>   ``BUILD_TESTING`` option to ``ON``.

Reviewer: S: If you do leave this as an entry in the 4.0 section, you can reword here to something like:
```suggestion
- Fixes a bug where auto-downloaded versions of the C driver set the ``ENABLE_TESTS`` configuration option to ``ON``, overriding the {+driver-short+} default value. ``ENABLE_TESTS`` is now set to ``OFF`` by default in all cases.
```

Reviewer: Also, if this is not a breaking change I'd just add it to What's New and not Upgrade

- Drops support for the ``MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX`` configuration
  option. Default CMake behavior for ``CMAKE_INSTALL_PREFIX`` is now respected.

.. _version-4.0-enable-tests-config:

- Sets the ``ENABLE_TESTS`` configuration option to ``OFF`` by default. To
  re-enable building test targets, you must set ``ENABLE_TESTS`` to ``ON``.
  To include test targets in the "all" target, you must also set the
  ``BUILD_TESTING`` option to ``ON``.

  .. note:: 

>     This change does not apply to auto-downloaded versions of the
>     {+driver-short+}. :ref:`The change for auto-downloaded versions was added in

Reviewer: I:
```suggestion
    This change does not apply to auto-downloaded versions of the
    C driver. :ref:`The change for auto-downloaded versions was added in
```

- Drops support for the ``MONGOCXX_OVERRIDE_DEFAULT_INSTALL_PREFIX`` configuration
  option. Default CMake behavior for ``CMAKE_INSTALL_PREFIX`` is now respected.

> .. note:: 

Reviewer: I: I think we should still leave this as a v4.0 change, since that release did set ``ENABLE_TESTS`` configuration option to ``OFF`` by default - the only time it didn't was if you auto-downloaded the C driver. and (from my understanding) that's the specific case the 4.1 release fixed

Reviewer 2: This is correct. The bugfix applies specifically to auto-downloaded C Driver configurations.

   The :ref:`ENABLE_TESTS configuration option change
   <version-4.1-enable-tests-config>` was originally announced for v4.0, but

>    instead was released with v4.1.

Reviewer 2: This is incorrect. The 4.0 entry should be restored as it was prior, although perhaps with a note that it does not apply to auto-downloaded C Driver configurations as intended due to a bug.
